offer-manager-domain/src/main/groovy/life/qbic/business/offers/OfferContent.groovy

```diff
@@ -0,0 +1,300 @@
+            customerStreet = customerAffiliation.street
+            customerPostalCode = customerAffiliation.postalCode
+            customerCity = customerAffiliation.city
+            customerCountry = customerAffiliation.country
+            /*Projectmanager*/
+            projectManagerFirstName = projectManager.firstName
+            projectManagerLastName = projectManager.lastName
+            projectManagerTitle = projectManager.title == AcademicTitle.NONE ? "" : projectManager.title
+            projectManagerEmail = projectManager.emailAddress
+
+            /*Projectinformation*/
+            this.creationDate = creationDate
+            this.expirationDate = expirationDate
+            this.projectTitle = projectTitle
+            this.projectObjective = projectObjective
+            this.experimentalDesign = experimentalDesign
+            this.offerIdentifier = offerIdentifier
+
+        }
+        Builder dataGenerationItems(List<ProductItem> dataGenerationItems){
+            this.dataGenerationItems = dataManagementItems
+            return this
+        }
+        Builder dataAnalysisItems(List<ProductItem> dataAnalysisItems){
+            this.dataAnalysisItems = dataAnalysisItems
+            return this
+        }
+        Builder dataManagementItems(List<ProductItem> dataManagementItems){
+            this.dataManagementItems = dataManagementItems
+            return this
+        }
+        Builder overheadTotal(double overheadTotal){
+            this.overheadTotal = overheadTotal
+            return this
+        }
+        Builder overheadsDataGeneration(double overheadDG){
+            this.overheadsDataGeneration = overheadDG
+            return this
+        }
+        Builder overheadDataAnalysis(double overheadDA){
+            this.overheadsDataAnalysis = overheadDA
+            return this
+        }
+        Builder overheadsPMandDS(double overheadPMandDS){
+            this.overheadsPMandDS = overheadPMandDS
+            return this
+        }
+        Builder netDataGeneration(double net){
+            this.netDataGeneration = net
+            return this
+        }
+        Builder netDataAnalysis(double net){
+            this.netDataAnalysis = net
+            return this
+        }
+        Builder netPMandDS(double net){
+            this.netPMandDS = net
+            return this
+        }
+        Builder totalCost(double total){
+            this.totalCost = total
+            return this
+        }
+        Builder netCost(double net){
+            this.netCost = net
+            return this
+        }
+        Builder totalVat(double vat){
+            this.totalVat = vat
+            return this
+        }
+
+        OfferContent build(){
+            return new OfferContent(this)
+        }
+    }
+
+    private OfferContent(Builder builder){
+        /*Person Information*/
+        /*Customer*/
+        customerFirstName = builder.customerFirstName
```

Comment:
I think you should add non-null requirements here.

Reply:
Indeed, very good point